python/mxnet/gluon/model_zoo/model_store.py

```diff
@@ -103,16 +105,19 @@ def get_model_file(name, root=os.path.join(base.data_dir(), 'models')):
 
     util.makedirs(root)
 
-    zip_file_path = os.path.join(root, file_name+'.zip')
     repo_url = os.environ.get('MXNET_GLUON_REPO', apache_repo_url)
     if repo_url[-1] != '/':
         repo_url = repo_url + '/'
-    download(_url_format.format(repo_url=repo_url, file_name=file_name),
-             path=zip_file_path,
-             overwrite=True)
-    with zipfile.ZipFile(zip_file_path) as zf:
-        zf.extractall(root)
-    os.remove(zip_file_path)
+
+    with tempfile.NamedTemporaryFile(dir=root) as zip_file:
+        download(_url_format.format(repo_url=repo_url, file_name=file_name),
+                 path=zip_file.name, overwrite=True, inplace=True)
```

Comment:
From the doc:

>  Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later). If delete is true (the default), the file is deleted as soon as it is closed.

Here you are taking the file name of the opened file, passing it to download and therein opening it again. You need to work with the open file object itself when using `tempfile.NamedTemporaryFile`.

Reply:
Thanks for the comment! That makes me think NamedTemporaryFile and TemporaryFile are quite restricted. I have to modify other functions that uses this file so that they operate on the open file object itself. For now I'll fallback to the previous method where the file is explicitly cleaned up.

Reply:
Yes. You can set `delete=False` and this restriction should go away.

Reply:
I think @apeforest suggested using `TemporaryFile` because it can be automatically cleaned up. If we need to cleanup ourselves, then there's not much difference using it or not